documentation/prometheus-design-principles.md

```diff
@@ -0,0 +1,164 @@
+# Prometheus design principles
+
+What is prometheus for?  There seem to be a number of design
+principles behind prometheus that aren't explicitly documented on the
+website.  However, prometheus maintainers sometimes refer to them in
+mailing lists or github issues.  This page tries to reverse engineer
+them.
+
+## Terminology
+
+The Prometheus community uses certain words as jargon terms.  These
+are some:
+
+  * failure domain: a group of things that can all fail at once (a
+    data centre, an AWS region or AZ)
+  * instance: an endpoint that you can scrape
+  * job: a collection of *instances* with the same purpose, replicated
+    for scalability or reliability or some other purpose
+
+## What Prometheus does
+
+Prometheus is a tool for collecting time-series data.
```

Comment:
These time-series can be queried using PromQL, a custom query language for time-series data. PromQL queries can be used as a basis for monitoring systems through dashboards and alerting.